File: tdd-kit/tdd-kit-ruby/lib/fizzbuzz.rb
class Fizzbuzz
  def elements
    (1..100).map(&method(:value_of))
  end

  def value_of(number)
    return 'fizzbuzz' if number % 15 == 0
    return 'fizz' if number % 3 == 0
    return 'buzz' if number % 5 == 0
    number.to_s
  end
end

File: tdd-kit/tdd-kit-java/src/main/java/Fizzbuzz.java
import com.google.common.collect.Lists;

import java.util.List;

public class Fizzbuzz {
    public List<String> elements() {
        List<String> elements = Lists.newArrayList();
        for (int i = 1; i <= 100; i++) {
            elements.add(valueOf(i));
        }
        return elements;
    }

    private String valueOf(int number) {
        if (number % 15 == 0) {
            return "fizzbuzz";
        }
        if (number % 3 == 0) {
            return "fizz";
        }
        if (number % 5 == 0) {
            return "buzz";
        }
        return String.valueOf(number);
    }
}
